### DISCUSSION TITLE
[New Feature] Scalable sidebar and radar

### DISCUSSION THREAD

**Discussion by Crimsonum** [Vinifera-Developers/Vinifera#1292] [Feature Requests]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, videos, links, etc)*.

### Description

Tiberian Sun does not adjust well to modern resolutions and widescreen displays. The sidebar and radar suffer especially, as they are fixed in size and location and do not scale with resolution.

Below is just a few of the issues with the sidebar on high resolution:
- increased distance between the extremes of the tactical view and the sidebar, increasing the time spent moving the cursor back and forth
- sidebar buttons and icons do not scale, requiring higher accuracy
- the radar display is too small to be useful, even on mediocre resolutions like 720p

Some of these issues can be mitigated with hotkeys, but hotkeys should never be a requirement for fluid gameplay (right now they are).

From what I know, most people these days play on either 1080p or 720p. Any lower, and the game feels claustrophobic and low definition (due to individual pixels becoming too apparent). Anything higher, and graphical game elements like infantry sprites become unbearably small.

Having the sidebar and radar scale with resolution, even if only to an extent, would be a major quality-of-life improvement to the game.

**Comment by CCHyper** [comment]
Until we can find a way to replace the core renderer with a modern interface _(OpenGL, SDL, DirectX9+, etc)_, this will not be possible. Using a new renderer interface would allow us to detach the tactical view and sidebar to separate layers, allowing the sidebar section to be scaled as required. It would also allow tactical view zooming to be done correctly.

Filed under "No Timeline" for now.

**Comment by Crimsonum** [comment]
It appears someone has been able to modernize the sidebar through DLL (ddraw?) magic, [see this thread for more information](https://ppmforums.com/topic-65817/command-new-world-order-preview-videos/). It does not seem like the sidebar has been upscaled, but there are other impressive feats like toggleable sidebar and transparent radar. Too bad the creator has not given out much information about the project yet.
